server/src/main/java/org/opensearch/common/geo/GeoUtils.java

```diff
@@ -74,6 +75,9 @@ public class GeoUtils {
     public static final String LONGITUDE = "lon";
     public static final String GEOHASH = "geohash";
 
+    public static final String GEOJSON_TYPE = "type";
+    public static final String GEOJSON_TYPE_POINT = "Point";
```

Comment:
Making a note here about case sensitivity. The GeoJson Specification ([RFC 7946; section 1.4](https://www.rfc-editor.org/rfc/rfc7946)) defines the geometry type strings as case-sensitive. The inherited GeoJson Parser, however, [converts every type string to uppercase](https://github.com/opensearch-project/OpenSearch/blob/77cff55bd431c0e2f8fdea5a40c83f8e8bddeca1/libs/geo/src/main/java/org/opensearch/geometry/ShapeType.java#L53) - I presume to make working with the enum easier. Technically this means our DSL is non-compliant with the GeoJson specification.  

Now, we don't necessarily require strict compliance with the specification but I do think we should be consistent so that if parsing `geo_point` as GeoJson fails w/ case sensitivity, so should parsing `geo_shape` fail with case sensitivity. 

So I think in this PR we should reuse  `ShapeType.forName` (see my inline suggestion below) to parse the shape type and discuss the case sensitivity compliance in a follow up issue (since it would be a breaking change).

Reply:
Thanks for pointing it out. Will make a change accordingly.